Inverted-Index/src/Operation.java

```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+abstract class Operation {
+    
+    protected Map<String, List<Doc>> map;
+    protected List<String> keyWords;
+    protected Set<Doc> docs;
```

Comment:
I couldn't find where this field (docs) is used. it's never initialized and never used.